pyaedt/edb_core/edb_data/siwave_simulation_setup_data.py

```diff
@@ -851,184 +1072,86 @@ def use_si_settings(self):
         -------
         bool
         """
-        return self._edb_sim_setup_info.SimulationSettings.UseSISettings
+        return self.get_sim_setup_info.SimulationSettings.UseSISettings
 
     @use_si_settings.setter
     def use_si_settings(self, value):
-        self._edb_sim_setup_info.SimulationSettings.UseCustomSettings = False
-        self._edb_sim_setup_info.SimulationSettings.UseSISettings = value
+        edb_setup_info = self.get_sim_setup_info
+        edb_setup_info.SimulationSettings.UseSISettings = value
+        self._edb_object = self._set_edb_setup_info(edb_setup_info)
         self._update_setup()
 
-    @pyaedt_function_handler()
-    def add_frequency_sweep(self, name=None, frequency_sweep=None):
-        """Add frequency sweep.
 
-        Parameters
-        ----------
-        name : str, optional
-            Name of the frequency sweep.
-        frequency_sweep : list, optional
-            List of frequency points.
-
-        Returns
-        -------
-        :class:`pyaedt.edb_core.edb_data.simulation_setup_data.EdbFrequencySweep`
-
-        Examples
-        --------
-        >>> setup1 = edbapp.create_siwave_syz_setup("setup1")
-        >>> setup1.add_frequency_sweep(frequency_sweep=[
-        ...                           ["linear count", "0", "1kHz", 1],
-        ...                           ["log scale", "1kHz", "0.1GHz", 10],
-        ...                           ["linear scale", "0.1GHz", "10GHz", "0.1GHz"],
-        ...                           ])
-        """
-        if name in self.frequency_sweeps:
-            return False
-        if not name:
-            name = generate_unique_name("sweep")
-        sweep = EdbFrequencySweep(self, frequency_sweep, name)
-        self._sweep_data_list[name] = sweep
-        return sweep
-
-
-def _parse_value(v):
-    """
+class SiwaveDCSimulationSetup(SiwaveSYZSimulationSetup):
+    """Manages EDB methods for SIwave DC simulation setup.
 
     Parameters
     ----------
-    v :
-
-
-    Returns
-    -------
-
+    pedb : :class:`pyaedt.edb.Edb`
+        Inherited AEDT object.
+    edb_setup : Ansys.Ansoft.Edb.Utility.SIWDCIRSimulationSettings
+        Edb object.
```

Comment:
```suggestion
        EDB object. The default is ``None``.
```
Describe the behavior that occurs when ``None`` is used.:The default is ``None``, in which case ....